File: backend/ecom/src/main/java/com/mursalin/ecom/service/AdminChatService.java
package com.mursalin.ecom.service;

import com.mursalin.ecom.dto.admin.chat.AdminChatSessionDetailDTO;
import com.mursalin.ecom.dto.admin.chat.AdminChatSessionSummaryDTO;
import com.mursalin.ecom.dto.admin.chat.AdminChatStatsDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.LocalDateTime;

public interface AdminChatService {
    Page<AdminChatSessionSummaryDTO> getSessions(String search, LocalDateTime startDate, LocalDateTime endDate,
                                                  Boolean hasUser, boolean escalatedOnly, Pageable pageable);

    AdminChatSessionDetailDTO getSessionDetail(Long id);

    AdminChatStatsDTO getStats();
}

File: frontend/ecom-frontend-5-main/src/services/adminChatService.js
import API from '../axios';

export const getChatSessions = async ({ search, startDate, endDate, escalatedOnly, hasUser, page = 0, pageSize = 20 }) => {
  const params = {
    page,
    pageSize,
    ...(search ? { search } : {}),
    ...(startDate ? { startDate } : {}),
    ...(endDate ? { endDate } : {}),
    ...(escalatedOnly !== undefined ? { escalatedOnly } : {}),
    ...(hasUser !== undefined ? { hasUser } : {}),
  };
  const response = await API.get('/admin/chat/sessions', { params });
  return response.data;
};

export const getChatSessionDetail = async (id) => {
  const response = await API.get(`/admin/chat/sessions/${id}`);
  return response.data;
};

export const getChatStats = async () => {
  const response = await API.get('/admin/chat/stats');
  return response.data;
};
